### PR TITLE
feat/registration-update

### PR DESCRIPTION
**Pull request by puffitos**
## Motivation

Part of https://github.com/caas-team/sparrow/issues/101

I'll split this into two PRs to keep them easy to merge. First comes the cleanup and refactoring needed and then comes the feature itself. 

## Changes

I've added a more general `Config` struct to the `targetManager`, which includes the general config each `targetManager` should have. This will also include a new field in a coming PR, to regulate the update interval.

I've added a simple Validation function for the general config, to prerpare for the new feature (0 duration will disable the update & registration). A zero checkInterval makes no sense, so it's not allowed.

The rest ist fixing tests and moving the configs to the `targets` package, as we've done with other PRs.

For additional information look at the commits.

## Tests done

- [x] New unittests

## Future plans

We can regulate the update & registration interval by using the magic number 0 (which is a possible value for the interval). Since the practice to use that number is common, we should adopt it, instead of just dropping the value as not-OK.

## TODO

- [x] I've assigned this PR to myself
- [x] I've labeled this PR correctly

### CODE REVIEW

File: pkg/sparrow/targets/targetmanager_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package targets
```
Comment (lvlcn-t) [inline pkg/sparrow/targets/targetmanager_test.go:1]: add license header

File: pkg/sparrow/targets/errors.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package targets
```
Comment (lvlcn-t) [inline pkg/sparrow/targets/errors.go:1]: add license header

**Review by lvlcn-t — APPROVED** [review]
you can add the license headers also in #104, so LGTM 😄

File: pkg/sparrow/targets/gitlab.go [hunk 3]
```diff
@@ -80,8 +76,8 @@ func (t *gitlabTargetManager) Reconcile(ctx context.Context) error {
 	log := logger.FromContext(ctx)
 	log.Info("Starting global gitlabTargetManager reconciler")
 
-	checkTimer := time.NewTimer(t.checkInterval)
-	registrationTimer := time.NewTimer(t.registrationInterval)
+	checkTimer := time.NewTimer(t.cfg.CheckInterval)
+	registrationTimer := time.NewTimer(t.cfg.RegistrationInterval)
```
Comment (NiklasTreml) [inline pkg/sparrow/targets/gitlab.go:80]: Can you add a unit test for the 0 case, where the update should never happen? I'm not 100% sure but I think a zero timer returns at least immediately, which would cause the opposite of what we want, 1000s of refreshes every second

**Review by NiklasTreml — CHANGES_REQUESTED** [review]

File: pkg/sparrow/targets/gitlab.go (hunk 3 above)
Comment (NiklasTreml) [reply pkg/sparrow/targets/gitlab.go:80]: looked into https://github.com/golang/go/issues/10124, you've already implemented there, very good 👍

**Review by NiklasTreml — APPROVED** [review]

**Review by lvlcn-t — APPROVED** [review]
